app/src/main/java/com/alexchan/wallpaper/ui/wallpaper/dashboard/dashboard/DashboardFragment.kt

```diff
@@ -0,0 +1,261 @@
+        val sharedPreferences: SharedPreferences = requireActivity().getSharedPreferences(
+            DISPLAY_VIEW_TYPE, Context.MODE_PRIVATE)
+
+        // Set Layout Manager Programmatically
+        when (sharedPreferences.getInt(DISPLAY_VIEW_TYPE_KEY, 2)) {
+            0 -> binding?.photosGrid?.layoutManager = StaggeredGridLayoutManager(1, StaggeredGridLayoutManager.VERTICAL)
+            1 -> binding?.photosGrid?.layoutManager = GridLayoutManager(requireContext(), 2)
+            else -> binding?.photosGrid?.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)
+        }
+
+        // Experimental
+        binding?.photosGrid?.adapter?.stateRestorationPolicy = RecyclerView.Adapter.StateRestorationPolicy.PREVENT_WHEN_EMPTY
+
+        // Add previous and next button on click listener and set current page number
+        if (MainActivity.showPagination) {
+            binding?.pageNumberTextView?.apply {
+                visibility = View.VISIBLE
+                text = MainActivity.pageNumber.toString()
+            }
+            binding?.previousButton?.apply {
+                visibility = View.VISIBLE
+                setOnClickListener {displayPreviousPage()}
+            }
+            binding?.nextButton?.apply {
+                visibility = View.VISIBLE
+                setOnClickListener {displayNextPage()}
+            }
+        }
+
+        binding?.photosGridSwipeRefresh?.isEnabled = false
+        // Handle Swipe Refresh Layout
+        if (MainActivity.showPagination) {
+            binding?.photosGridSwipeRefresh?.apply {
+                when (resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) {
+                    Configuration.UI_MODE_NIGHT_YES -> {
+                        setProgressBackgroundColorSchemeResource(R.color.colorOnPrimary)
+                        setColorSchemeResources(R.color.colorPrimary)
+                    }
+                    Configuration.UI_MODE_NIGHT_NO -> {
+                        setProgressBackgroundColorSchemeResource(R.color.colorPrimary)
+                        setColorSchemeResources(R.color.colorOnPrimary)
+                    }
+                }
+                isEnabled = true
+                setOnRefreshListener {refreshDashboard()}
+            }
+        }
+    }
+
+    private fun refreshDashboard() {
+        // Get current internet connection status
+        val connectionManager = requireActivity().getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager?
+        val networkManager = connectionManager?.activeNetwork
+        val activeNetwork = connectionManager?.getNetworkCapabilities(networkManager)
+        // Check if it is connected to mobile data or wifi and have valid connected access
+        if (activeNetwork != null) {
+            if (activeNetwork.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ||
+                activeNetwork.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) ||
```

Comment:
This should be moved to service layer and can be reuse for other screen do network call.